### PR TITLE
First pass at adding nullspace bias tasks for the velocity and position solvers.

### PR DESCRIPTION
**Pull request by rethink-forrest**
Added a secondary task to push the IK solutions toward a desired joint bias in the nullspace. 

This has not been fully optimized, but I am pushing this for further testing and integration with the rest of the system. The velocity solver saw a time increase of ~40% while the position solvers time increased 2-3x. The IK solvers should be able to recognize tasks in the joint configuration space to reduce unnecessary matrix math operations. The nullspace bias changes were not pushed into the testing framework.

I'm am open to suggestions for improvements to the interfaces.

### CODE REVIEW

File: sns_ik_examples/src/ik_tests.cpp [hunk 1]
```diff
@@ -213,12 +214,13 @@ void test(ros::NodeHandle& nh, double num_samples_pos, double num_samples_vel,
...
     result = JointSeed[i];
     start_time = boost::posix_time::microsec_clock::local_time();
+    int cnt = 0;
     do {
       q=result; // when iterating start with last solution
       rc=kdl_solver.CartToJnt(q,end_effector_pose,result);
       diff = boost::posix_time::microsec_clock::local_time() - start_time;
       elapsed = diff.total_nanoseconds() / 1e9;
-    } while (rc < 0 && elapsed < timeout);
+    } while (rc < 0 && elapsed < timeout && cnt++ < 100);
```
Comment (rethink-imcmahon) [inline sns_ik_examples/src/ik_tests.cpp]: What is this count variable simulating? Probably should have a comment somewhere in here.

Comment (rethink-forrest) [inline sns_ik_examples/src/ik_tests.cpp]: It's just enforcing a max number of iterations under 100.

File: sns_ik_lib/src/sns_position_ik.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ SNSPositionIK::SNSPositionIK(KDL::Chain chain, std::shared_ptr<SNSVelocityIK> ve
     m_jacobianSolver(chain),
     m_linearMaxStepSize(0.2),
     m_angularMaxStepSize(0.2),
-    m_maxIterations(150),
+    m_maxIterations(100),
```
Comment (rethink-imcmahon) [inline sns_ik_lib/src/sns_position_ik.cpp]: Any reason for the value change?

Comment (rethink-forrest) [inline sns_ik_lib/src/sns_position_ik.cpp]: The solver will now fail faster. I tested different max iterations, and found that anything beyond 100 didn't really increase the success rate but increased the average run time.

File: sns_ik_lib/include/sns_ik/sns_ik.hpp [hunk 3]
```diff
@@ -77,8 +78,51 @@ namespace sns_ik {
...
-    int CartToJnt(const KDL::JntArray& q_in, const KDL::Twist& v_in, KDL::JntArray& qdot_out);
+    inline bool getJointNames(std::vector<std::string> jointNames) {
+      jointNames = m_jointNames;
+      return m_initialized;
+    }
+
+    int CartToJnt(const KDL::JntArray &q_init,
+                  const KDL::Frame &p_in,
+                  KDL::JntArray &q_out,
+                  const KDL::Twist& tolereances=KDL::Twist::Zero())
```
Comment (rethink-imcmahon) [inline sns_ik_lib/include/sns_ik/sns_ik.hpp]: typo

**Comment by rethink-imcmahon** [comment]
Once you make these changes, you have my :shipit: to merge

**Comment by rethink-forrest** [comment]
I've addressed the feedback from Ian and checked that it builds with stricter build warnings.

**Pull request merged**